### PR TITLE
Update PR template and remove PULL_REQUEST_TEMPLATE directory

### PR DESCRIPTION
**Pull request by rrzhang139**
### Description

Removed PULL_REQUEST_TEMPLATE subdirectory and added extra sections in PR template. 

### Issue Link

N/A

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Contents

- remove PULL_REQUEST_TEMPLATE directory
- Update PR template

### Rationale

We removed PULL_REQUEST_TEMPLATE due to a [bug](https://stackoverflow.com/questions/52139192/github-pull-requests-template-not-showing/64367212#64367212).
We want to link the issue with the PR it is addressing. Also, we added a test section so we know how it was tested. 

### How Has This Been Tested?

N/A

### CODE REVIEW

**Comment by rrzhang139** [comment]
#1075

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -1,42 +0,0 @@
...
-
-<hr>
-
-## How to fill a PR description (remove this)
-
-Please give a concise description of your PR.
-
-The target readers could be future developers, reviewers, and auditors. By reading your description, they should easily understand the changes proposed in this pull request.
-
-MUST: Reference the issue to resolve
```
Comment (ChihChengLiang) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:20]: Can we add back the guidance on how to write the description?

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -1,42 +0,0 @@
...
-- Added xxxx function to ...
-- Refactored ....
-```
-
-### Design choices
-
-RECOMMENDED to:
-- What types of design choices did you face?
-- What decisions you have made?
-- Any valuable information that could help reviewers to think critically
```
Comment (ChihChengLiang) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:42]: We extensively discussed how the PR description could best help the reviewing process. We concluded that it is most helpful when the PR author discloses more detail about the design choices. Can we add back this part?

**Review by ChihChengLiang — COMMENTED** [review]
I like some descriptions of the original template, can we add those back?

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md (hunk 1 above)
Comment (rrzhang139) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md]: Definitely. Thanks for the correction.

**Review by ChihChengLiang — APPROVED** [review]
LGTM. Pending review from Edu

**Review by ed255 — APPROVED** [review]
LGTM!

**Pull request merged**